src/pages/Settings/Settings.js

```diff
@@ -1,146 +1,182 @@
+import { sliceHash as presentHash } from 'utils'
+import HashAvatar from 'components/HashAvatar'
+import Modal from 'components/Modal'
 import PrimaryLayout from 'components/layout/PrimaryLayout'
 import Button from 'components/Button'
-import Input from 'components/Input'
+import HposSettingsQuery from 'graphql/HposSettingsQuery.gql'
+import HposStatusQuery from 'graphql/HposStatusQuery.gql'
+import HposUpdateVersionMutation from 'graphql/HposUpdateVersionMutation.gql'
 
-import './Settings.module.css'
+export const createLabelfromSnakeCase = string => string.replace(/([A-Z])/g, ' $1').split(' ').map(word => capitalize(word)).join(' ')
+
+const DEFAULT_PORT_NAMES = ['Device Admin', 'HC Network', 'Hosting']
+const NOT_AVAILABLE = 'Not Available'
 
-const portValidationRules = yup.number()
-  .typeError('Port must be specified.') // TypeError because empty value gets cast into NaN
-  .min(1000, 'Ports must be between 1000 and 65000.')
-  .max(65000, 'Ports must be between 1000 and 65000.')
-  .required()
-const SettingsValidationSchema = yup.object().shape({
-  hostName: yup.string().required(),
-  hostPubKey: yup.string().required(),
-  registrationEmail: yup.string()
-    .email()
-    .required(),
-  deviceName: yup.string().required(),
-  networkId: yup.string().required(),
-  deviceAdminPort: portValidationRules,
-  hcAdminPort: portValidationRules,
-  hcNetworkPort: portValidationRules,
-  hostingPort: portValidationRules
-})
-
-export function Settings ({
-  settings,
-  updateSettings,
-  history: { push },
-  toggleSshAccess
-}) {
-  const { register, handleSubmit, errors } = useForm({
-    defaultValues: settings,
-    validationSchema: SettingsValidationSchema
+// Data - Mutation hook
+function useUpdateVersion () {
+  const [hposUpdateVersion] = useMutation(HposUpdateVersionMutation)
+  return (availableVersion) => hposUpdateVersion({
+    variables: { availableVersion }
   })
+}
 
-  const [sshAccessVal, setSshAccess] = useState(false)
-  const handleToggleSshAccess = (e) => {
-    e.preventDefault()
-    setSshAccess(e.target.checked)
-    toggleSshAccess()
-  }
+export function Settings ({ history: { push } }) {
+  const { data: { hposSettings: settings = [] } = {} } = useQuery(HposSettingsQuery)
+  const { data: { hposStatus: status = [] } = {} } = useQuery(HposStatusQuery)
+
+  const updateVersion = useUpdateVersion()
+  const [softwareUpdateVersion, setSoftwareUpdateVersion] = useState()
+  const showSoftwareUpdateModal = availableVersion => setSoftwareUpdateVersion(availableVersion)
 
-  const onSubmit = settings => {
-    updateSettings(settings)
+  let updateAvailable = false
+  let availableVersion, currentVersion
+  if (!isEmpty(status) && !isEmpty(status.versionInfo)) {
+    availableVersion = status.versionInfo.availableVersion
+    currentVersion = status.versionInfo.currentVersion
   }
+  if (!isEmpty(availableVersion) && !isEmpty(currentVersion) && (availableVersion !== currentVersion)) updateAvailable = true
```

Comment:
`const updateAvailable = (!isEmpty(availableVersion) && !isEmpty(currentVersion) && (availableVersion !== currentVersion))`

avoids having to predefine  `updateAvailable` on line 35